File: 2025-12-18_week03/A/solutions/solution.py
def fractured_ledger(N):
    # محاسبه phi برای همه اعداد تا N با sieve
    phi = list(range(N + 1))
    for i in range(2, N + 1):
        if phi[i] == i:  # i عدد اول است
            for j in range(i, N + 1, i):
                phi[j] -= phi[j] // i

    # مجموع phi ها
    total_phi = sum(phi[1:])

    # مجموع phi برای اعداد فرد
    odd_phi = sum(phi[i] for i in range(1, N + 1, 2))

    # فرمول نهایی
    return 2 * total_phi - 1 + odd_phi


# ورودی
N = int(input())
print(fractured_ledger(N))

File: 2025-12-18_week03/A/solutions/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;

    vector<int> phi(N + 1);
    for (int i = 1; i <= N; i++)
        phi[i] = i;

    for (int i = 2; i <= N; i++) {
        if (phi[i] == i) { // i عدد اول است
            for (int j = i; j <= N; j += i) {
                phi[j] -= phi[j] / i;
            }
        }
    }

    long long total_phi = 0;
    long long odd_phi = 0;

    for (int i = 1; i <= N; i++) {
        total_phi += phi[i];
        if (i & 1) odd_phi += phi[i];
    }

    long long answer = 2 * total_phi - 1 + odd_phi;
    cout << answer << "\n";

    return 0;
}
